homepage/homepage/start.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+## Create git SSH key
+SSH_DIR="/app/config/ssh"
+SSH_KEYFILE="${SSH_DIR}/id_rsa"
+SSH_KNOWNHOSTS_FILE="${SSH_DIR}/known_hosts"
+test -f "${SSH_KEYFILE}" && export GIT_SSH_COMMAND="ssh -i '${SSH_KEYFILE}' -o UserKnownHostsFile='${SSH_KNOWNHOSTS_FILE}'"
+
+## Add known SSH host keys:
+touch "${SSH_KNOWNHOSTS_FILE}"
+if ! grep github.com "${SSH_KNOWNHOSTS_FILE}" > /dev/null
+then
+     echo "github.com ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCj7ndNxQowgcQnjshcLrqPEiiphnt+VTTvDP6mHBL9j1aNUkY4Ue1gvwnGLVlOhGeYrnZaMgRK6+PKCUXaDbC7qtbW8gIkhL7aGCsOr/C56SJMy/BCZfxd1nWzAOxSDPgVsmerOBYfNqltV9/hWCqBywINIR+5dIg6JTJ72pcEpEjcYgXkE2YEFXV1JHnsKgbLWNlhScqb2UmyRkQyytRLtL+38TGxkxCflmO+5Z8CSSNY7GidjMIZ7Q4zMjA2n1nGrlTDkzwDCsw+wqFPGQA179cnfGWOWRVruj16z6XyvxvjJwbz0wQZ75XK5tKSb7FNyeIEs4TT4jk+S4dhPeAUC5y+bDYirYgM4GC7uEnztnZyaVWQ7B381AK4Qdrwt51ZqExKbQpTUNn+EjqoTwvqNj4kqx5QUCI0ThS/YkOxJCXmPUWZbhjpCg56i+2aB6CmK2JGhn57K5mj0MNdBXA4/WnwH6XoPWJzK5Nyu2zB3nAZp+S5hpQs+p1vN1/wsjk=" >> "${SSH_KNOWNHOSTS_FILE}"
+fi
+REPO_DOMAIN=$(echo "${HOMEPAGE_TEMPLATE_REPO}" | grep -oP '(?<=@|:\/\/)([^\/:]+)' | sed 's/[^@]*@//')
+REPO_PORT=$(echo "${HOMEPAGE_TEMPLATE_REPO}" | grep -oP '(?<=:)\d+')
+if [[ ! -z "${REPO_PORT}" ]]; then
+    GREP_REPO="[${REPO_DOMAIN}]:${REPO_PORT}"
+else
+    GREP_REPO="${REPO_DOMAIN}"
+fi
+if ! grep "$(echo "${GREP_REPO}" | sed 's/\./\\./g; s/\[/\\[/g; s/\]/\\]/g')" "${SSH_KNOWNHOSTS_FILE}" > /dev/null
```

Comment:
This line appears to be adding backslashes github.com -> github\.com .... but why?

Reply:
Please add some commentary before all the sed lines.

Reply:
```
## Pull domain from HOMEPAGE_TEMPLATE_REPO, which can be in 1 of 2 formats:
## git@github.com:YourUsername/my-private-homepage-template.git
## https://github.com/EnigmaCurry/d.rymcg.tech_homepage-template.git
REPO_DOMAIN=$(echo "${HOMEPAGE_TEMPLATE_REPO}" | grep -oP '(?<=@|:\/\/)([^\/:]+)' | sed 's/[^@]*@//')
## Pull port from HOMEPAGE_TEMPLATE_REPO.
## I'm not setting the default port of 22 if no port is designated in
## HOMEPAGE_TEMPLATE_REPO because when a domain without a specified port
## is added to known_hosts, it's added without a port - I'm trying
## to grep for the HOMEPAGE_TEMPLATE_REPO in known_hosts, so if it doesn't
## have a port there, I don't want to grep for a port.
REPO_PORT=$(echo "${HOMEPAGE_TEMPLATE_REPO}" | grep -oP '(?<=:)\d+')
if [[ ! -z "${REPO_PORT}" ]]; then
    GREP_REPO="[${REPO_DOMAIN}]:${REPO_PORT}"
else
    GREP_REPO="${REPO_DOMAIN}"
fi
## I'm adding backslashes before . [ and ] in the domain because when I grep
## for HOMEPAGE_TEMPLATE_REPO, these characters need to be escaped.
if ! grep "$(echo "${GREP_REPO}" | sed 's/\./\\./g; s/\[/\\[/g; s/\]/\\]/g')" "${SSH_KNOWNHOSTS_FILE}" > /dev/null
then
    ssh-keyscan -t rsa -p "${REPO_PORT}" "${REPO_DOMAIN}" 2>/dev/null | grep -v '^#' >> "${SSH_KNOWNHOSTS_FILE}"
fi
```

Reply:
cleaned up greps, removed most seds